rust/datafusion/src/physical_plan/string_expressions.rs

```diff
@@ -70,6 +71,16 @@ pub fn concatenate(args: &[ArrayRef]) -> Result<StringArray> {
     Ok(builder.finish())
 }
 
+/// extract a specific group from a string column, using a regular expression
+pub fn regexp_extract(args: &[ArrayRef]) -> Result<ArrayRef> {
+    let pattern_expr = args[1].as_any().downcast_ref::<StringArray>().unwrap();
+    let pattern = pattern_expr.value(0);
+    let idx_expr = args[2].as_any().downcast_ref::<Int64Array>().unwrap();
+    let idx = idx_expr.value(0) as usize;
```

Comment:
I am not sure this is correct: the function expects an array, but then only picks the first element of the array for the regex. Maybe this was used because ScalarFunctions did not support the `ScalarValue` variant?

Reply:
Absolutely, I did this because I am not sure, how I can get from `ArrayRef` to `ScalarVariant` - I am looking for something like the following:

```
    let pattern_expr = args[1].as_any().downcast_ref::<ScalarValue>().unwrap();
    if let ScalarValue::Utf8(Some(pattern)) = pattern_expr {
        compute::regexp_match(args[0].as_ref(), pattern)
            .map_err(DataFusionError::ArrowError)
    } else {
        Err(DataFusionError::Internal("This is wrong".to_string()))
    }
```